bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/binding/BaseThingHandler.java

```diff
@@ -392,13 +392,25 @@ protected void updateThing(Thing thing) {
             throw new IllegalArgumentException(
                     "Changes must not be done on the current thing - create a copy, e.g. via editThing()");
         }
+        ThingHandlerCallback callback = this.callback;
+        if (callback == null) {
+            logger.warn("Handler {} tried updating thing {} although the handler was already disposed.",
+                    this.getClass().getSimpleName(), thing.getUID());
+            return;
+        }
+        try {
+            callback.validateConfigurationParameters(thing, thing.getConfiguration().getProperties());
+        } catch (ConfigValidationException e) {
+            logger.warn(
+                    "Attempt to update thing '{}' with a thing containing invalid configuration '{}', blocked. This is most likely a bug.",
+                    thing.getUID(), thing.getConfiguration());
+            return;
+        }
         synchronized (this) {
             if (this.callback != null) {
```

Comment:
This check can be omitted by reusing the local variable `callback` defined earlier.